File: dayone/Day1.cs
using System;
using System.Collections;
using System.IO;
using System.Linq;


class Day1 {
    static void Main(string[] args) {
        partone();
        parttwo();
    }

    static void partone() {
        int last = 0, ans = 0;
        bool first = true;
        foreach (string line in File.ReadAllLines("Day1.in")) {
            int x = Convert.ToInt32(line);
            if (first) {
                first = false;
                last = x;
                continue;
            }
            if (x > last) ++ans;
            last = x;
        }
        Console.WriteLine(ans);
    }

    static void parttwo() {
        int ans = 0;
        int[] tmp = File.ReadAllLines("Day1.in").Select(int.Parse).ToArray();
        int sum = tmp[0] + tmp[1] + tmp[2];
        for (int i = 3; i < tmp.Length; ++i) {
            int x = sum - tmp[i-3] + tmp[i];
            if (x > sum) ++ans;
            sum = x;
        }
        Console.WriteLine(ans);
    }
}

File: dayone/Day1.cpp
#include "bits/stdc++.h"
using namespace std;

void partone(bool console=false), parttwo(bool console=false);

int main() {
    partone(1);
    parttwo(1);
    return 0;
}

void partone(bool console) {
    ifstream fin = ifstream("Day1.in");
    ofstream fout = ofstream("Day1.out");
    string s;
    vector<int> v;
    while (fin >> s) {
        v.emplace_back(stoi(s));
    }
    int ans = 0;
    for (int i = 1; i < v.size(); ++i) {
        if (v[i] >= v[i-1]) ++ans;
    }
    (console?cout:fout)<< ans << '\n';
    fout.close();
}


void parttwo(bool console) {
    ifstream fin = ifstream("Day1.in");
    ofstream fout = ofstream("Day1.out");
    string s;
    vector<int> v;
    while (fin >> s) {
        v.emplace_back(stoi(s));
    }
    int ans = 0;
    long long sum = 0;
    sum += v[0] + v[1] + v[2];
    for (int i = 3; i < v.size(); ++i) {
        long long newsum = sum - v[i-3] + v[i];
        if (newsum > sum) ++ans;
        sum = newsum;
    }
    (console?cout:fout) << ans << '\n';
    fout.close();
}


